lib/ddtrace/profiling/ext/cpu.rb

```diff
@@ -19,6 +20,65 @@ def self.apply!
           # will provide a thread/clock ID for CPU timing.
           require 'ddtrace/profiling/ext/cthread'
           ::Thread.send(:prepend, Profiling::Ext::CThread)
+
+          # Applying hooks will allow any existing threads to update
+          # their thread/clock IDs so their CPU timings can be measured.
+          apply_hooks!
+        end
+
+        def self.apply_hooks!
+          # Threads that have already been created, will not have resolved
+          # a thread/clock ID. This is because these IDs can only be resolved
+          # from within the thread's execution context, which we do not control.
+          #
+          # We can work around this by applying trace hooks to capture each thread
+          # from within its own execution context, resolving the thread/clock IDs,
+          # then disabling the hook so it doesn't re-run needlessly.
+          Thread.list.each do |thread|
+            # Only attempt to hook into threads that are instrumented
+            next unless thread.is_a?(CThread)
```

Comment:
I'm still a bit unsure that we need this check, e.g. see the discussion we started in https://github.com/DataDog/dd-trace-rb/pull/1310/files#r561027708 .

Reply:
It shouldn't be necessary, but I think this is an example of inexpensive defensive programming. The potential benefit (not having errors in the hook) outweighs the cost (doing a check.) Given the sensitivity of this code, I would avoid being assumptive of preconditions.

Reply:
Hmm I guess. I agree with defensive programming, but in this case I would rather just wrap everything with a begin/rescue -- that way we could catch every thing we haven't thought of, rather than just this specific case.